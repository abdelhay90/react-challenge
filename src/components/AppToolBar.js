import React from 'react';
import Typography from '@material-ui/core/Typography';
import { AppBar, makeStyles, Toolbar } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  appBar: {
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
  toolbar: {
    flexWrap: 'wrap',
  },
  toolbarTitle: {
    flexGrow: 1,
  },
  link: {
    margin: theme.spacing(1, 1.5),
  },
}));

export default function AppToolBar() {
  const classes = useStyles();
  return (
    <AppBar
      position='static'
      color='default'
      elevation={0}
      className={classes.appBar}
    >
      <Toolbar className={classes.toolbar}>
        <Typography
          variant='h6'
          color='inherit'
          noWrap
          className={classes.toolbarTitle}
        >
          SWVL
        </Typography>
      </Toolbar>
    </AppBar>
  );
}
