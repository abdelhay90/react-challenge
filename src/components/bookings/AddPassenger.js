import React, { useState } from 'react';
import { inject } from 'mobx-react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Fab,
  Tooltip,
} from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import PassengerForm from './PassengerForm';

const AddPassenger = inject('trip')(({ trip }) => {
  const [open, setOpen] = useState(false);
  const [addLabel, setAddLabel] = useState('Add Passenger');
  const toggleOpen = () => {
    setOpen(!open);
    if (trip.isTripMaxCapacity) {
      setAddLabel('Max Capacity');
    }
  };
  const disableClickAndEsc = true;
  return (
    <>
      <Tooltip title={addLabel} aria-label='add'>
        <Fab
          size='small'
          aria-label='Add Passenger'
          color='primary'
          onClick={toggleOpen}
          disabled={trip.isTripMaxCapacity}
        >
          <AddIcon />
        </Fab>
      </Tooltip>
      <Dialog
        fullWidth
        maxWidth='md'
        open={open}
        disableBackdropClick={disableClickAndEsc}
        disableEscapeKeyDown={disableClickAndEsc}
        onClose={toggleOpen}
      >
        <DialogTitle id='max-width-dialog-title'>
          Add Customer to Current Trip
        </DialogTitle>
        <DialogContent>
          <PassengerForm onPassengerAdded={toggleOpen} />
        </DialogContent>
        <DialogActions>
          <Button onClick={toggleOpen} color='primary'>
            Close
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
});

export default AddPassenger;
