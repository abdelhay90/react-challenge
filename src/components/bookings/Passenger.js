/**
 * Passenger card that hold the passenger identity, status, and payment method
 */
import React from 'react';
import Box from '@material-ui/core/Box';
import { Avatar, makeStyles, Typography } from '@material-ui/core';

const useStyles = makeStyles({
  passenger: {
    minWidth: 340,
    maxWidth: 340,
  },
  bigAvatar: {
    margin: 1,
    width: 60,
    height: 60,
    border: '1px solid rgba(88,88,88,0.1)',
    'box-shadow': '2px 0px rgba(88,88,88,0.3)',
  },
});

export default function Passenger({
  id,
  name,
  status,
  paymentMethod,
  imageUrl,
}) {
  const classes = useStyles();
  return (
    <Box
      id={id}
      display='flex'
      flexWrap='wrap'
      p={1}
      mx={1}
      className={classes.passenger}
    >
      <Box p={1} mx={1} alignItems='center'>
        <Avatar alt={name} src={imageUrl} className={classes.bigAvatar}>
          {imageUrl === '' ? name[0] : ''}
        </Avatar>
      </Box>
      <Box p={1} mx={1} xs={6}>
        <Typography variant='h6' gutterBottom>
          {name}
        </Typography>
        {`${status} - ${paymentMethod}`}
      </Box>
    </Box>
  );
}
