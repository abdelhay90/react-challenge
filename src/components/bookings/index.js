export Bookings from './Bookings';
export PassengerList from './PassengerList';
export Passenger from './Passenger';
export AddPassenger from './AddPassenger';
export PassengerForm from './PassengerForm';
