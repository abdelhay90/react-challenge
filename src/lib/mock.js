export const stops = [
  {
    stationId: '5d865ee507bfe196bac94ff3',
    stationName: 'Talaat Harb Axis',
    coordinates: [29.995097117000945, 31.445976804476345].reverse(),
  },
  {
    stationId: '5d865ee543724f1184914907',
    stationName: 'Omar Ibn Abdelaziz',
    coordinates: [30.012880226989733, 31.431022685763764].reverse(),
  },
  {
    stationId: '5d865ee54ec4f96a85fa3dbc',
    stationName: 'Southern Teseen',
    coordinates: [30.016423452010145, 31.43356512343098].reverse(),
  },
  {
    stationId: '5d865ee54fs346a85fa3dbc',
    stationName: 'AlMoshir Tantawy Axis',
    coordinates: [30.016375545253084, 31.39839364118893].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a85fa3dbc',
    stationName: 'El-Shaikh Kamel Khedr',
    coordinates: [30.040987895343214, 31.34613896298947].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a86trh6bc',
    stationName: 'Ahmed ElZomor',
    coordinates: [30.04505095423728, 31.34143552756268].reverse(),
  },
  {
    stationId: '432756654ec4f96a85fa3dbc',
    stationName: 'Mostafa ElNahas',
    coordinates: [30.054089469238654, 31.34217149725646].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a85fa3987',
    stationName: 'Makram Ebid',
    coordinates: [30.0619822837204, 31.34523262262769].reverse(),
  },
  {
    stationId: '5d8734554ec4f96a85fa3dbc',
    stationName: 'Makrem Ebeid Ext',
    coordinates: [30.073225027534153, 31.343785066473743].reverse(),
  },
  {
    stationId: '5d87566598u4f96a85fa3dbc',
    stationName: 'ElThawra St.',
    coordinates: [30.08201257054215, 31.343882169980134].reverse(),
  },
];

export const passengers = [
  {
    id: '5d865ee507bfe196bac94ff3',
    name: 'Bailey Walls',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee543724f1184914907',
    name: 'Angie Rice',
    status: 'MISSED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee54ec4f96a85fa3dbc',
    name: 'Cecilia Trevino',
    status: 'CANCELLED',
    paymentMethod: 'credit',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5ad0a247f264dd75f',
    name: 'Coffey Fleming',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5ecfd4d136af665ba',
    name: 'Stuart Orr',
    status: 'CANCELLED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee5b3da2fb19e62a897',
    name: 'Vickie Beard',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee525b0567a26b1c2f3',
    name: 'Sandra Blackburn',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee5f18d6587c1cb9ed0',
    name: 'Tia Oneal',
    status: 'CANCELLED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee527d6024df9300d92',
    name: 'Marylou Carr',
    status: 'MISSED',
    paymentMethod: 'CREDIT',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee51049173f6c386500',
    name: 'Rowe Rocha',
    status: 'MISSED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5d0d97ea49d4e5124',
    name: 'Mcconnell Jefferson',
    status: 'COMPLETED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d8734554ec4f96a85fa3dbc',
  },
  // {
  //   id: '5d865ee580c33a45ed14c2ed',
  //   name: 'Cook Best',
  //   status: 'CANCELLED',
  //   paymentMethod: 'CREDIT',
  //   imageUrl: '/static/images/full-face-1.jpg',
  //   pickupStationId: '5d865ee507bfe196bac94ff3',
  //   checkoutStationId: '5d87566598u4f96a85fa3dbc'
  // },
];

export const trip = {
  id: '123456789',
  distance: 47,
  rateFare: 25,
  driverInfo: {
    carModel: ' TOYOTA HiAce 2019',
    carImageUrl: '/static/images/Toyota-HiAce.jpg',
    driverName: 'Ahmed Abdelhay',
    driverImageUrl: '/static/images/driver-1.jpg',
    plateNumber: 'QA 2378',
  },
  path: stops.slice(0),
  passengers: passengers.slice(0),
};
