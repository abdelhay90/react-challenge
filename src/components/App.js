import React from 'react';
import 'mapbox-gl/dist/mapbox-gl.css';
import { Container, CssBaseline, makeStyles } from '@material-ui/core';
import AppToolBar from './core/AppToolBar';
import {
  MainMapContainer,
  TripInfoContainer,
  BookingsContainer,
} from '../containers';

const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
    ul: {
      margin: 0,
      padding: 0,
    },
    li: {
      listStyle: 'none',
    },
  },
  mainContainer: {
    padding: theme.spacing(8, 0, 6),
  },
  mapContainer: {
    'padding-top': '100px',
    'padding-right': '16px',
    'padding-left': '16px',
  },
}));

export default function App() {
  const classes = useStyles();
  return (
    <>
      <CssBaseline />
      <AppToolBar />
      <Container
        maxWidth='lg'
        component='main'
        className={`${classes.mainContainer} ${classes.mapContainer}`}
      >
        <MainMapContainer />
      </Container>
      <Container
        maxWidth='lg'
        component='main'
        className={classes.mainContainer}
      >
        <TripInfoContainer />
      </Container>
      <Container
        maxWidth='lg'
        component='main'
        className={classes.mainContainer}
      >
        <BookingsContainer />
      </Container>
    </>
  );
}
