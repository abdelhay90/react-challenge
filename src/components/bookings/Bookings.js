import React from 'react';
import { Box, Typography } from '@material-ui/core';
import Passenger from './Passenger';

function Bookings({ passengers = [] }) {
  return (
    <>
      <Box px={1} mx={1}>
        <Typography variant='h5'>Bookings</Typography>
      </Box>
      <Box display='flex' flexWrap='wrap' p={1} m={1}>
        {passengers.map(passenger => (
          <Passenger
            id={passenger.id}
            name={passenger.name}
            status={passenger.name}
            paymentMethod={passenger.name}
            imageUrl={passenger.imageUrl}
          />
        ))}
      </Box>
    </>
  );
}

export default Bookings;
