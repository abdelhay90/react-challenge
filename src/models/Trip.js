import { observable, action, computed } from 'mobx';
import groupBy from 'lodash/groupBy';
import Passenger from './Passenger';
import {
  PASSENGER_STATUS,
  PASSENGERS_MAX_CAPACITY,
  TRIP_STATUS,
} from '../lib/constants';

export default class Trip {
  id;

  @observable distance;

  @observable rateFare;

  @observable driverInfo;

  @observable path;

  @observable passengers = [];

  @observable tripStatus;

  @observable currentStop;

  constructor({
    id,
    passengers = [],
    distance,
    rateFare,
    driverInfo,
    path,
    tripStatus = TRIP_STATUS.NOT_STARTED,
  }) {
    this.id = id;
    this.passengers = passengers.map(item => new Passenger({ ...item }, this));
    this.distance = distance;
    this.rateFare = rateFare;
    this.driverInfo = driverInfo;
    this.path = path;
    this.currentStop = this.path.length > 0 ? this.path[0] : null;
    this.tripStatus = tripStatus;
  }

  @computed
  get route() {
    return {
      name:
        this.path.length !== 0
          ? `${this.path[0].stationName} - ${this.path[this.path.length - 1].stationName}`
          : 0,
      route: this.path.map(item => item.coordinates),
      routeStops: this.path,
      startPoint: this.path.length !== 0 ? this.path[0] : '',
      endPoint: this.path.length !== 0 ? this.path[this.path.length - 1] : '',
      currentStop: this.currentStop,
    };
  }

  @computed
  get isTripMaxCapacity() {
    return this.passengers.length >= PASSENGERS_MAX_CAPACITY;
  }

  @computed
  get passengersStats() {
    let allGroups = Object.keys(PASSENGER_STATUS).reduce(
      (accum, value) => ({ ...accum, [value]: [] }),
      {},
    );
    const calculatedGroups = groupBy(this.passengers, 'status');
    allGroups = { ...allGroups, ...calculatedGroups };
    return Object.keys(allGroups).map(key => ({
      x: key,
      y: allGroups[key].length,
    }));
  }

  getStationMap() {
    return this.path.reduce(
      (accum, value) => ({ ...accum, [value.stationId]: [] }),
      {},
    );
  }

  combineStationStats(allStations, calculated) {
    const newStatsMap = { ...allStations, ...calculated };
    return Object.keys(newStatsMap).map(key => ({
      x: this.path.filter(item => item.stationId === key)[0].stationName,
      y: newStatsMap[key].length,
    }));
  }

  @computed
  get stationPickupStats() {
    const allStations = this.getStationMap();
    const passengers = this.passengers.filter(
      passenger =>
        passenger.status !== PASSENGER_STATUS.MISSED &&
        passenger.status !== PASSENGER_STATUS.CANCELLED,
    );
    const stationGroups = groupBy(passengers, 'pickupStationId');
    return this.combineStationStats(allStations, stationGroups);
  }

  @computed
  get stationCheckoutStats() {
    const allStations = this.getStationMap();
    const passengers = this.passengers.filter(
      passenger => passenger.status === PASSENGER_STATUS.COMPLETED,
    );
    const stationGroups = groupBy(passengers, 'checkoutStationId');
    return this.combineStationStats(allStations, stationGroups);
  }

  @action.bound
  addPassenger(passenger) {
    if (this.passengers.length < 12)
      this.passengers.push(new Passenger({ ...passenger }, this));
  }

  @action.bound
  removePassenger(item) {
    this.passengers = this.passengers.filter(i => i !== item);
  }

  @action.bound
  startTrip() {
    this.tripStatus = TRIP_STATUS.STARTED;
  }

  @action.bound
  stopTrip() {
    this.tripStatus = TRIP_STATUS.FINISHED;
  }
}
