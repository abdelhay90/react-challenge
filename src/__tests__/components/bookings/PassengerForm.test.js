import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import Trip from '../../../models/Trip';
import { trip } from '../../../lib/mock';
import { PassengerForm } from '../../../components/bookings';

describe('Passenger Form Component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    const currentTrip = new Trip({ ...trip });
    ReactDOM.render(
      <Provider trip={currentTrip}>
        <PassengerForm />
      </Provider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should match snapshot', () => {
    const currentTrip = new Trip({ ...trip });
    const tree = renderer
      .create(
        <Provider trip={currentTrip}>
          <PassengerForm />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change values in inputs', () => {
    const currentTrip = new Trip({ ...trip });
    const wrapper = mount(
      <Provider trip={currentTrip}>
        <PassengerForm />
      </Provider>,
    );
    const evt = new window.Event('input', { bubbles: true });
    evt.simulated = true;
    wrapper
      .find("input[id='name']")
      .last()
      .getDOMNode().value = 'ahmed abdelhay';
    wrapper
      .find("input[id='name']")
      .last()
      .simulate('change');
    wrapper
      .find("input[id='mobileNumber']")
      .last()
      .getDOMNode().value = '010042651856';
    wrapper
      .find("input[id='mobileNumber']")
      .last()
      .simulate('change');
    expect(
      wrapper
        .find("input[id='name']")
        .last()
        .getDOMNode().value,
    ).toEqual('ahmed abdelhay');
    expect(
      wrapper
        .find("input[id='mobileNumber']")
        .last()
        .getDOMNode().value,
    ).toEqual('010042651856');
  });
});
