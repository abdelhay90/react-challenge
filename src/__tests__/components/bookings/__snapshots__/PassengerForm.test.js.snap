// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passenger Form Component should match snapshot 1`] = `
<div
  className="makeStyles-paper-217"
>
  <form
    className="makeStyles-form-219"
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="name"
          >
            Customer Name
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="cname"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-382 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-383"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="mobileNumber"
          >
            Mobile Number
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="mobileNumber"
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="mobileNumber"
              name="mobileNumber"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-382 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-383"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-select-pickup-station"
          >
            Select
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
              id="select-pickupStationId"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-describedby="outlined-select-pickup-station-helper-text"
              id="outlined-select-pickup-station"
              name="pickupStationId"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-382 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-383"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="outlined-select-pickup-station-helper-text"
          >
            Please select pick up station
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-select-checkout-station"
          >
            Select
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
              id="select-checkoutStationId"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-describedby="outlined-select-checkout-station-helper-text"
              id="outlined-select-checkout-station"
              name="checkoutStationId"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-382 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-383"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="outlined-select-checkout-station-helper-text"
          >
            Please select checkout station
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="outlined-select-payment"
          >
            Select
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
              id="select-paymentMethod"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              CASH
            </div>
            <input
              aria-describedby="outlined-select-payment-helper-text"
              id="outlined-select-payment"
              name="paymentMethod"
              type="hidden"
              value="CASH"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-382 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-383"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
            id="outlined-select-payment-helper-text"
          >
            Please select payment method
          </p>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-submit-220 MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Add Booking
      </span>
    </button>
  </form>
</div>
`;
